FreeRTOS/Demo/CORTEX_R4_RM46_HERCULES_MCU_GCC/CMakeLists.txt

```diff
@@ -0,0 +1,194 @@
+cmake_minimum_required(VERSION 3.9)
+
+SET(CMAKE_CROSSCOMPILING "TRUE" CACHE STRING "Set Cross Compiling to true" FORCE)
+
+# Strip the default MacOSX flags that cause cross-compilations to fail.
+SET(CMAKE_OSX_DEPLOYMENT_TARGET "" CACHE STRING "Force unset of the deployment target for iOS" FORCE)
+SET(CMAKE_OSX_SYSROOT "" CACHE STRING "Force unset of the deployment target for iOS" FORCE)
+
+# Set the compiler before declaring the project for the test build
+SET(CMAKE_C_COMPILER "arm-none-eabi-gcc")
+SET(CMAKE_ASM_COMPILER "arm-none-eabi-gcc")
+
+# Set the system processor and name before declaring the project
+# Needs to be set here otherwise it will fail the test compilation
+SET(CMAKE_SYSTEM_NAME "Generic" CACHE STRING "Target system is a generic ARM Processor")
+SET(CMAKE_SYSTEM_PROCESSOR "armv7-r" CACHE STRING "Target system is an ARM7r Processor")
+
+# Set the ASM and C compilation flags
+SET(CMAKE_ASM_FLAGS "-mcpu=cortex-r4 -mfpu=vfpv3-d16 -Og -g -ggdb -Wall -MMD -MP")
+SET(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} -specs=\"nosys.specs\" -specs=\"nano.specs\"")
+SET(CMAKE_C_FLAGS "${CMAKE_ASM_FLAGS} -marm -mfloat-abi=hard")
+
+project(RM46_FreeRTOS C ASM)
+cmake_minimum_required(VERSION 3.5)
+
+SET(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR} CACHE STRING "")
+
+# Increase the debug level of the CMAKE build
+SET(CMAKE_VERBOSE_MAKEFILE ON)
+
+# Get the absolute path to the Demo Directory
+SET(DEMO_DIR_REL "${CMAKE_CURRENT_SOURCE_DIR}")
+GET_FILENAME_COMPONENT(DEMO_DIR ${DEMO_DIR_REL} ABSOLUTE)
+
+# Get the absolute path to the Board Files
+SET(BOARD_FILES_DIR_REL "${DEMO_DIR}/BoardFiles")
+GET_FILENAME_COMPONENT(BOARD_FILES_DIR ${BOARD_FILES_DIR_REL} ABSOLUTE)
+
+SET(FREERTOS_CONFIG_FILE_DIRECTORY "${DEMO_DIR}/include" CACHE STRING "Config File Path")
+SET(FREERTOS_PORT "GCC_ARM_CRX_MPU" CACHE STRING "FreeRTOS Port to Use")
+
+# Clone the tag of the FreeRTOS-Kernel last tested with this project.
+INCLUDE(FetchContent)
+
+FetchContent_Declare(
```

Comment:
Going off @bradleysmith23's earlier comments, FetchContent was not added to CMake until version 3.11, so the minimum should at least be 3.11. It's probably best practice to try to run the CMakeLists with whatever version is listed as the minimum and increment from there, but probably not that big of an issue.

Reply:
Thanks for catching that, I'll update that in the next push to the demo